src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,108 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+public class MovieRecommender {
+    private String file;
+    private Hashtable<String, Integer> HashProduct = new Hashtable<String, Integer>();
+    private Hashtable<Integer, String> InvertedHashProduct = new Hashtable<Integer, String>();
+    private Hashtable<String, Integer> HashUser = new Hashtable<String, Integer>();
+    private int users =0, products =0, reviews = 0;
+
+    public MovieRecommender(String file) throws IOException{
+        this.file = file;
+        getData();
+    }
+
+    public String getData() throws IOException {
+        int thisProduct =0, thisUser =0;
+        Files.deleteIfExists(Paths.get("Result.csv"));
+        File result = new File("Result.csv");
+        InputStream fileReader = new GZIPInputStream(new FileInputStream(this.file));
+        BufferedReader br = new BufferedReader(new InputStreamReader(fileReader));
+        FileWriter fileWriter = new FileWriter(result);
+        BufferedWriter bw = new BufferedWriter(fileWriter);
+        String line;
+        String[] sp;
+        String key, value;
+
+        while((line = br.readLine()) != null) {
+            if (line.length() >= 0) {
+                sp = line.split(" ");
+                key = sp[0];
+                if (key.equals("product/productId:")) {
```

Comment:
`key.equals("product/productId:")` can produce a null pointer exception, 
`"product/productId:".equals(key)` can't, something to have in mind.